romancal/regtest/test_wfi_grism_pipeline.py

```diff
@@ -0,0 +1,156 @@
+
+
+@pytest.fixture(scope="module")
+def output_filename(run_elp):
+    return run_elp.output
+
+
+@pytest.fixture(scope="module")
+def output_model(output_filename):
+    with rdm.open(output_filename) as model:
+        yield model
+
+
+@pytest.fixture(scope="module")
+def truth_filename(run_elp):
+    return run_elp.truth
+
+
+@pytest.fixture(scope="module")
+def repointed_filename_and_delta(output_filename):
+    delta = [10.0, 10.0]
+
+    with rdm.open(output_filename) as model:
+        # open from the file since we're going to modify the model
+        del model.meta["wcs"]
+
+        # Create new pointing for the model
+        # RA & Dec are each shifted + 10 degrees, unless they are near
+        # the upper limit, in which case they are shifted -10 degrees.
+        if model.meta.wcsinfo.ra_ref >= 350.0:
+            delta[0] *= -1.0
+        if model.meta.wcsinfo.dec_ref >= 80.0:
+            delta[1] *= -1.0
+
+        model.meta.wcsinfo.ra_ref += delta[0]
+        model.meta.wcsinfo.dec_ref += delta[1]
+
+        # Create new wcs object for the new pointing
+        model = AssignWcsStep.call(model)
+
+        # save repointed model
+        repointed_filename = output_filename.rsplit(".", 1)[0] + "_repoint.asdf"
+        model.to_asdf(repointed_filename)
+
+    return repointed_filename, delta
+
+
+def test_output_matches_truth(output_filename, truth_filename, ignore_asdf_paths):
+    # DMS9 There was no test of the ephemeris so I guess this covered it?
+    diff = compare_asdf(output_filename, truth_filename, **ignore_asdf_paths)
+    assert diff.identical, diff.report()
+
+
+@pytest.mark.parametrize(
+    "step_name, status",
+    (
+        ("assign_wcs", "COMPLETE"),
+        ("dark", "COMPLETE"),
+        ("dq_init", "COMPLETE"),
+        ("jump", "COMPLETE"),
+        ("linearity", "COMPLETE"),
+        ("ramp_fit", "COMPLETE"),
+        ("saturation", "COMPLETE"),
+        ("refpix", "COMPLETE"),
+        # skipped
+        ("flat_field", "SKIPPED"),
+        ("photom", "SKIPPED"),
+        ("source_detection", "SKIPPED"),
+        ("tweakreg", "SKIPPED"),
+    ),
+)
+def test_step_status(output_model, step_name, status):
+    # DMS90
+    # DMS278
+    # also DMS129 for assign_wcs
+    assert getattr(output_model.meta.cal_step, step_name) == status
+
+
+def test_jump_in_uneven_ramp(output_model):
+    # DMS365 jump detection detected jumps in uneven ramp
+    uneven = len({len(x) for x in output_model.meta.exposure.read_pattern}) > 1
+    assert uneven & np.any(output_model.dq & pixel.JUMP_DET)
+
+
+@pytest.mark.parametrize("arr_name", ("dq", "err", "var_poisson", "var_rnoise"))
+def test_array_exists(output_model, arr_name):
+    # DMS91
+    assert hasattr(output_model, arr_name)
+
+
+def test_has_exposure_time(output_model):
+    # DMS88 total exposure time exists
+    assert "exposure_time" in output_model.meta.exposure
+
+
+def test_wcs_has_bounding_box(output_model):
+    # DMS93 WCS tests  TODO isn't listed
```

Comment:
DMS93 isn't listed in the requirements. Is it satisfied and should it be?

Reply:
I think the issue here is that DMS9 is spuriously listed as being tested in this step, and we should have done DMS93.

Reply:
Thanks! Clear as mud :)

I removed DMS9 and added DMS93 in 03ed9b4